keps/0009-operator-toolkit.md

```diff
@@ -0,0 +1,296 @@
+      - backup-pv.yaml
+  backup:
+    resources:
+      - base-job.yaml
+    patches:
+      - backup.yaml
+  restore:
+    resources:
+      - restore.yaml
+  load-data:
+    resources:
+      - job.yaml
+  clear-data:
+    resources:
+      - job.yaml
+  query:
+    resources:
+      - job.yaml
+plans:
+  deploy:
+    steps:
+      - name: deploy
+        tasks:
+          - deploy
+      - name: init
+        tasks:
+          - init
+      - name: cleanup
+        tasks:
+          - init
+        delete: true
+  backup:
+    triggers:
+      - parameter: backupFile
+    steps:
+      - name: pv
+        tasks:
+          - pv
+      - name: backup
+        tasks:
+          - backup
+      - name: cleanup
+        tasks:
+          - backup
+        delete: true
+  restore:
+    steps:
+      - name: restore
+        tasks:
+          - restore
+      - name: cleanup
+        tasks:
+          - restore
+        delete: true
+```
+
+While subsequent sections go into deeper detail, the top level keys of the operator are:
+
+- **version**: String defining the version of a given operator
+- **tasks**: A map of tasks that can be run. These are the atomic runnable unit of a KUDO operator, and are made up of a series of YAML manifests. These are defined more in detail in [Tasks](#tasks).
+- **plans**: A map of plans that can be run. These are the core partitioning unit of a KUDO operator. A plan is intended to run a single "operations task" for an operator, such as backup, restore, deployment, or update. This is defined in detail in [Plans](#plans)
+
+This file undergoes a Go template pass on Instance instantiation before being parsed. This is described more in detail in [Extensions and Bases](#extensions-and-bases)
+
+#### params.yaml
+
+The `params.yaml` file is a struct that defines parameters for operator. This can articulate descriptions, defaults, and triggers etc., In the MySQL example, this looks like:
+
+```yaml
+backupFile:
+  description: "The name of the backup file"
+  default: backup.sql
+password:
+  default: password
+  description: "Password for the mysql instance"
+```
+
+These values are meant to be overridden by Instance resources when instantiating an operator.
+
+This file undergoes a Go template pass on Instance instantiation before being parsed. This is described more in detail in [Parameters](#parameters).
+
+#### templates/
+
+The templates directory contains 0-to-many YAML manifests for operator tasks to use. These are described in more detail in [Templates](#templates).
+
+Each template undergoes a Go template pass for an Instance when it's corresponding Task is run.
+
+### Plans
+
+Plans are the core unit of operation within a KUDO operator. Each KUDO plan represents an individual unit of operations work. This may include backups, restores, deployments, upgrades, compaction, or any other myriad of operations that an application operator may want to perform with KUDO.
+
+A single plan is composed of [Steps](#steps).
```

Comment:
How about a small section explaining the parallelism of plans, steps, and tasks?
```suggestion
A single plan is composed of [Steps](#steps) and a step is composed of [tasks](#tasks). Individual steps are executed sequentially by default while the tasks within the step are executed in parallel.
```